ui/app/controllers/evaluations/index.js

```diff
@@ -1,12 +1,83 @@
 import Controller from '@ember/controller';
 import { action } from '@ember/object';
 import { tracked } from '@glimmer/tracking';
+import { schedule } from '@ember/runloop';
 import { inject as service } from '@ember/service';
+import { useMachine } from 'ember-statecharts';
+import { use } from 'ember-usable';
+import evaluationsMachine from '../../machines/evaluations';
 
 export default class EvaluationsController extends Controller {
+  @service store;
   @service userSettings;
 
-  queryParams = ['nextToken', 'pageSize', 'status'];
+  @tracked width = null;
+  @tracked height = null;
+
+  @use
+  statechart = useMachine(evaluationsMachine).withConfig({
+    services: {
+      loadEvaluation: this.loadEvaluation,
+    },
+    actions: {
+      updateEvaluationQueryParameter: this.updateEvaluationQueryParameter,
+      removeCurrentEvaluationQueryParameter:
+        this.removeCurrentEvaluationQueryParameter,
+    },
+    guards: {
+      sidebarIsOpen: this._sidebarIsOpen,
+    },
```

Comment:
Can you help me understand how `guards` is used? I'm not familiar with the pattern and I don't see many other examples of this pattern within Nomad. Is this a statecharts convention?

Reply:
The guard is being used to determine if the machine should be in the open state based on a query-param being set when the application gets loaded in this route - i.e. if you load the page with a "currentEval" query-param we know that this means the sidebar should be opened.